### DISCUSSION TITLE
Why does this line of code crash?

### DISCUSSION THREAD

**Discussion by shenshengyi** [vsg-dev/VulkanSceneGraph#1247] [General]
vsg::scale(2.0, 2.0, 2.0) * vsg::translate(1.0, 1.0, 1.0);

it's wired

**Comment by robertosfield** [comment]
It's all very straightforward maths code, not something we'd expect to crash. 

Could you provide the code in context and a stack trace for the crash you are seeing.

**Reply by shenshengyi** [reply]
auto scale_m = vsg::scale(2.0);
    auto scale_transform = vsg::MatrixTransform::create(scale_m);
    scale_transform->addChild(node);
    scene->addChild(scale_transform);

    scene->addChild(node);

    auto scale_translate_m4 = vsg::scale(2.0, 2.0, 2.0) * vsg::translate(1.0, 1.0, 1.0);****_（If you comment this line of code, it will be OK, otherwise it will crash.）**_**

    // auto transform = vsg::MatrixTransform::create(scale_translate_m4);
    // transform->addChild(node);
    // scene->addChild(transform);

    // create the viewer and assign window(s) to it
    vsg::ref_ptr<vsg::Viewer> viewer = vsg::Viewer::create();

    vsg::ref_ptr<vsg::Window> window(vsg::Window::create(windowTraits)); //

    viewer->addWindow(window);
    // compute the bounds of the scene graph to help position the camera
    vsg::ComputeBounds computeBounds;
    scene->accept(computeBounds);   _（ call stack crash here.）_



The debug run is fine.

The release run will crash.
It's strange, I just defined this variable and didn't use it.

The call stack shows the crash occurred elsewhere.

Failed to upload the image. Sorry, I have to describe it in text.

**Reply by robertosfield** [reply]
As a general comment, your original post was really unhelpful for others like myself.  From the comments in your code extract it's now clear the the original comment doesn't contain any code that crashes.

When you post questions please spend a bit more time thinking through the problem and how to convey what is actually going on, not some tiny fragment of code loosely associated with the actual problem.  If others are having to ask for you more information it's a sign that you've failed to communicate the issue properly - it's a lot more work for us to try and help you.

Often just the act of providing more information about the code that is failing can actually help you solve the problem yourself, or bring us all further along to helping you.

Still the lack of stack trace is still the main issue, we don't have your application, we don't have your data, we just have some code extracts that may or may not be associated with the issue.

**Reply by shenshengyi** [reply]
Thank you very much for your suggestion, I will check my configuration environment again.
